backend/app/services/internal/auth_service.py

```
@@ -0,0 +1,103 @@
+        if existing_user:
+            raise CustomHTTPException(status_code=400, detail="Email already registered")
+
+        # Hash the password
+        password_hash = get_password_hash(user_data.password)
+
+        # Create user
+        try:
+            user = self.user_repo.create_user(user_data, password_hash)
+            return UserResponse.from_orm(user)
+        except ValueError as e:
+            raise CustomHTTPException(status_code=400, detail=str(e))
```

Ruff (0.14.8)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)